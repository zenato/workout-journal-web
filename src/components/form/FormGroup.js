import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './FormGroup.scss';

const cx = classNames.bind(styles);

const FormGroup = ({ label, children }) => (
  <div className={cx('item')}>
    <label>{label}</label>
    <div className={cx('field')}>{children}</div>
  </div>
);

FormGroup.propTypes = {
  label: PropTypes.string,
  children: PropTypes.node,
};

export default FormGroup;
