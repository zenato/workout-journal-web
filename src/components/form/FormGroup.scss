@import 'common';

.item {
  display: flex;
  margin-bottom: 5px;
  align-items: center;

  label {
    flex: 2;
    text-align: right;
    font-weight: bold;
    margin-right: 5px;
  }

  .field {
    flex: 10;
    padding-right: 10px;
    vertical-align: middle;
  }
}
