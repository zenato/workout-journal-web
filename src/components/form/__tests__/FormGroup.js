import React from 'react';
import { shallow } from 'enzyme';
import FormGroup from '../FormGroup';

describe('FormGroup', () => {
  const props = {
    label: 'test',
  };

  it('should render passed props.', () => {
    const component = shallow(<FormGroup {...props}>Test</FormGroup>);
    expect(component).toMatchSnapshot();
  });
});
