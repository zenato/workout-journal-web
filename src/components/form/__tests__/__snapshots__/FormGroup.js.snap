// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should render passed props. 1`] = `
<div
  className="item"
>
  <label>
    test
  </label>
  <div
    className="field"
  >
    Test
  </div>
</div>
`;
